src/analyses/baseInvariant.ml

```diff
@@ -100,6 +100,25 @@ struct
         if M.tracing then M.tracel "inv" "st from %a to %a" D.pretty st D.pretty r;
         r
       )
+    | Mem (Lval lv), NoOffset ->
+      let lvals = eval_lv ~man st x in
+      let res = AD.fold (fun a acc -> 
+          if M.tracing then M.tracel "inv" "Consider case of lval %a = %a" d_lval lv VD.pretty (Address (AD.singleton a));
+          let st = set' lv (Address (AD.singleton a)) st in
+          let old_val = get ~man st (AD.singleton a) None in
+          let old_val = VD.cast (Cilfacade.typeOfLval x) old_val in (* needed as the type of this pointer may be different *)
+          (* this what I would originally have liked to do, but eval_rv_lval_refine uses queries and thus stale values *)
+          (* let old_val = eval_rv_lval_refine ~man st exp x in *)
+          let old_val = map_oldval old_val (Cilfacade.typeOfLval x) in
+          let v = apply_invariant ~old_val ~new_val:c' in
+          if is_some_bot v then 
+            D.join acc (try contra st with Analyses.Deadcode -> D.bot ())
```

Comment:
This now swallows all deadcode, but this might not be entirely right. In particular, if the invariant doesn't hold for any pointees, this will join together a bunch of `bot`s, but not raise `Deadcode`, which I think it would've done before.
We might have to have some extra logic here that if all cases were dead, then `Deadcode` is also raised after the fold. I think `MCP` does something similar (but other way around, for any instead of all).

Reply:
I fixed this and added a new regression test.